### PR TITLE
feat: Import changes from #1123 for Key Vault - `avm/res/key-vault/vault`

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
## Description

I have migrated changes from 1123 into my own feature branch to close and completes the related tasks to 1123
#closes 1122
- adds tests for ec and rsa kty key parameters
- adds checks for dependencies between keySize and curveName key parameters
- Expanded UDT for Secrets 
- Exapanded UDT for Keys
- added support for releasePolicy for key
- resolves the conflicts from #1123 

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.key-vault.vault](https://github.com/ChrisSidebotham/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)     |

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [x] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change (#1123 IS OPEN AND SHOULD BE DISCARDED)
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by ChrisSidebotham** [comment]
@fblix - if you get chance can you approve this if you get time. If we don't have a repsonse by the end of the week then the core team will approve this.

**Review by fblix — APPROVED** [review]

**Pull request merged**
